pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -572,12 +573,14 @@ func (a *Operator) ensureSingletonRBAC(operatorNamespace string, csv *v1alpha1.C
 			}
 			// TODO: this should do something smarter if the cluster role binding already exists
 			if crb, err := a.opClient.CreateClusterRoleBinding(clusterRoleBinding); err != nil {
-				// if the CR already exists, but the label is correct, the cache is just behind
-				if k8serrors.IsAlreadyExists(err) && ownerutil.IsOwnedByLabel(crb, csv) {
-					continue
-				} else {
+				if !k8serrors.IsAlreadyExists(err) {
 					return err
 				}
+				// if the CRB already exists, but the label is correct, the cache is just behind
+				if crb != nil && ownerutil.IsOwnedByLabel(crb, csv) {
+					continue
+				}
+				return err
```

Comment:
Should this error be returned?

Reply:
I think this needs to return an error if we want `!IsOwnedByLabel` to cause a resync (i.e. to retry this logic).